### ISSUE TITLE
Proxy authentication with certificate validation

### ISSUE THREAD

**Issue by rafal-dudek** [floragunncom/search-guard#697]
Hi,

I am using Elasticsearch with SearchGuard in Openshift.
I wanted to use "proxy_auth_domain" to authenticate my Grafana Users to Elasticsearch. Below configuration is working but is not secure. Any user can create request to Elasticsearch using header of any other User.

     proxy_auth_domain:
        http_enabled: true
        transport_enabled: true
        order: 3
        http_authenticator:
          type: proxy
          challenge: false
          config:
            user_header: "X-Grafana-User"
        authentication_backend:
          type: noop

I can not use `searchguard.dynamic.http.xff.internalProxies` to restrict ip address because it is dynamic.
It would be nice to have in proxy_auth_domain second authenticator e.g. certificate or basic internal user. Only combination of both authenticators (e.g. Grafana certificate and "X-Grafana-User" header with user name)  should grant permissions.

**Comment by jochenkressin** [comment]
May I ask why you want to use proxy authentication for Grafana? AFAIK Grafana supports Basic Auth out-of-the-box: https://docs.search-guard.com/latest/elasticsearch-grafana-search-guard

**Comment by rafal-dudek** [comment]
I want to grant different ElasticSearch privileges for different Grafana (Openshift) users. Setting Basic Auth in Grafana will grant same privileges for all Grafana Users.

**Comment by jochenkressin** [comment]
Ok, I see. Yes, that's a serious limitation in Grafana I have to admit. We already have a feature request in our backlog which would add TLS client certificates to the proxy authenticator. So instead of (or in combination with) the XFF settings, this would accept proxy auth only if a configured TLS client cert is sent with the request. Would that solve your issue?

**Comment by rafal-dudek** [comment]
If it would be one TLS client cert from proxy server (e.g. Grafana) working with different users in user_header (so users would have different privileges) it would solve our problems.

**Comment by jochenkressin** [comment]
Yes, that is exactly what we have in mind. IP based control is fine as long as you do not run on a dynamic environment like K8s. I will mark this issue here as feature request and also try to push it from the backlog to one of the next sprints.
